drivers/gpio/gpio_nrfx.c

```diff
@@ -315,271 +285,87 @@ static int gpio_nrfx_manage_callback(const struct device *port,
-						 NRF_GPIOTE_EVENT_PORT);
-
-	if (port_event) {
-		#define GPIO_NRF_GET_TRIGGERS(i) \
-			fired_triggers[DT_INST_PROP(i, port)] = \
-				check_level_trigger_pins(DEVICE_DT_INST_GET(i), \
-							 &sense_levels[DT_INST_PROP(i, port)]);
-
-		DT_INST_FOREACH_STATUS_OKAY(GPIO_NRF_GET_TRIGGERS)
-		#undef GPIO_NRF_GET_TRIGGERS
-
-		/* Sense detect was disabled while checking pins so
-		 * DETECT should be deasserted.
-		 */
-		nrf_gpiote_event_clear(NRF_GPIOTE, NRF_GPIOTE_EVENT_PORT);
-	}
-
-	/* Handle interrupt from GPIOTE channels. */
-	for (size_t i = 0; i < GPIOTE_CH_NUM; i++) {
-		nrf_gpiote_event_t evt =
-			offsetof(NRF_GPIOTE_Type, EVENTS_IN[i]);
-
-		if (nrf_gpiote_int_enable_check(NRF_GPIOTE, BIT(i)) &&
-		    nrf_gpiote_event_check(NRF_GPIOTE, evt)) {
-			uint32_t abs_pin = nrf_gpiote_event_pin_get(NRF_GPIOTE, i);
-			/* Divide absolute pin number to port and pin parts. */
-			fired_triggers[abs_pin / 32U] |= BIT(abs_pin % 32);
-			nrf_gpiote_event_clear(NRF_GPIOTE, evt);
-		}
-	}
-
-	if (IS_ENABLED(CONFIG_GPIO_NRF_INT_EDGE_USING_SENSE) && port_event) {
-		/* Reprogram sense to match current edge to be detected. Make it
-		 * now to detect all new edges after callbacks were fired.
-		 *
-		 * This may cause DETECT to be re-asserted if pin state has
-		 * already changed to the newly configured sense level.
-		 */
-		#define GPIO_NRF_CFG_EDGE_SENSE_PINS(i) \
-		   cfg_edge_sense_pins(DEVICE_DT_INST_GET(i), \
-				       sense_levels[DT_INST_PROP(i, port)]);
-
-		DT_INST_FOREACH_STATUS_OKAY(GPIO_NRF_CFG_EDGE_SENSE_PINS)
-		#undef GPIO_NRF_CFG_EDGE_SENSE_PINS
-	}
-
-	#define GPIO_NRF_FIRE_CALLBACKS(i) \
-		if (fired_triggers[DT_INST_PROP(i, port)]) { \
-			fire_callbacks(DEVICE_DT_INST_GET(i), \
-				       fired_triggers[DT_INST_PROP(i, port)]); \
-		}
-
-	DT_INST_FOREACH_STATUS_OKAY(GPIO_NRF_FIRE_CALLBACKS)
-	#undef GPIO_NRF_FIRE_CALLBACKS
-
-	if (port_event) {
-		/* Reprogram sense to match current configuration.
-		 * This may cause DETECT to be re-asserted.
-		 */
-		#define GPIO_NRF_CFG_LEVEL_PINS(i) cfg_level_pins(DEVICE_DT_INST_GET(i));
-
-		DT_INST_FOREACH_STATUS_OKAY(GPIO_NRF_CFG_LEVEL_PINS)
-		#undef GPIO_NRF_CFG_LEVEL_PINS
-	}
-}
-
-
 #define GPIOTE_NODE DT_INST(0, nordic_nrf_gpiote)
 
 static int gpio_nrfx_init(const struct device *port)
 {
 	static bool gpio_initialized;
+	nrfx_err_t err;
 
-	if (!gpio_initialized) {
-		gpio_initialized = true;
-		IRQ_CONNECT(DT_IRQN(GPIOTE_NODE), DT_IRQ(GPIOTE_NODE, priority),
-			    gpiote_event_handler, NULL, 0);
+	if (nrfx_gpiote_is_init()) {
+		return 0;
+	}
 
-		irq_enable(DT_IRQN(GPIOTE_NODE));
-		nrf_gpiote_int_enable(NRF_GPIOTE, NRF_GPIOTE_INT_PORT_MASK);
+	err = nrfx_gpiote_init(0/*not used*/);
+	if (err != NRFX_SUCCESS) {
+		return -EIO;
 	}
 
+	gpio_initialized = true;
+	nrfx_gpiote_global_callback_set(nrfx_gpio_handler, NULL);
+
+	IRQ_CONNECT(DT_IRQN(GPIOTE_NODE), DT_IRQ(GPIOTE_NODE, priority),
+		    nrfx_gpiote_irq_handler, NULL, 0);
```

Comment:
```suggestion
		    nrfx_isr, nrfx_gpiote_irq_handler, 0);
```
so that we stay consistent with other nrfx driver shims.